### DISCUSSION TITLE
Custom charge_extensions gives raise_validation_error

### DISCUSSION THREAD

**Discussion by Rubioli** [pay-rails/pay#521] [Q&A]
First of all, THANKS a lot for such a great gem!

In my application, I have an `Order` model and I'm trying to set up the `charge_extensions` just as your example, but I get:

```
pid=59586 tid=ows2z4q26 WARN: ActiveRecord::RecordInvalid: Validation failed: Order must exist
```
And  `raise_validation_error`

This is my `charge_extensions`: 

```
module ChargeExtensions
  extend ActiveSupport::Concern

  included do
    belongs_to :order
    after_create :fulfill_order
  end

  def fulfill_order
    order.fulfill!
  end
end
```

PS: do I need to have a column called `fulfill` in my `Order` model? 

I appreciate any help and thanks in advanced

**Comment by excid3** [comment]
As the error states, you've added an Order relation and not populated it.

**Comment by Rubioli** [comment]
I'm truly sorry if the question sounds bad, but I'm kind of new and confused here and would appreciate your help :)

I tried: 
```
class Order < ApplicationRecord
  has_many :pay_charges
```

and also:
```
class Order < ApplicationRecord
  has_many :charge_extensions
```

with no luck :/ could you please give me a hint :) Appreciate it

**Reply by excid3** [reply]
`belongs_to` are required unless you add `optional: true` so you have to assign the relationship.

**Reply by Rubioli** [reply]
Thanks a lot, @excid3 for your help. I was reading the guide and I saw this part: https://github.com/pay-rails/pay/blob/master/docs/stripe/1_overview.md#fulfilling-orders-after-checkout-completed which helps with my issue. I'm new to this and I'm not sure if I need to create a file for all if this and/or include it to a controller. Would you please help me out and shed some light on it. It would help me out of a huge problem. Thanks a lot :)
